lib/utils.js

```diff
@@ -31,6 +33,14 @@ var assign = Object.assign
       return to;
     };
 
+const uuidV4 = () => {
```

Comment:
a couple comments might be useful here too - at least for me, bit manipulation code is never really readable on its own. I had to spend some time reading the RFC spec to understand what these ops are doing